org-formation/040-budgets/cur.yaml

```diff
@@ -3,12 +3,33 @@ Description: 'Monthly Cost and Usage Report'
 Parameters:
   resourcePrefix:
     Type: String
+  bucketName:
+    Type: String
+  lambdaArn:
+    Type: String
 Resources:
   CostReportBucket:
     Type: "AWS::S3::Bucket"
     Properties:
-      BucketName: !Sub "${resourcePrefix}-cost-reports"
+      BucketName: !Ref bucketName
       AccessControl: BucketOwnerFullControl
+      NotificationConfiguration:
+        LambdaConfigurations:
+          - Event: "s3:ObjectCreated:*"
+            Function: !Ref lambdaArn
+            Filter:
+              S3Key:
+                Rules:
+                  - Name: suffix
+                    Value: ".snappy.parquet"
+  LambdaPermission:
```

Comment:
hmm, shouldn't the lambda permission go in the lambda template?

Reply:
I originally put this here to couple it with the event notification on the bucket, and even though I'm tempted to move the managed policy from the lambda template into this one to consolidate the bucket references in a single template, I agree that we should put as much lambda configuration into the lambda template as possible. I'll move this.